include/range/v3/view_interface.hpp

```diff
@@ -85,13 +85,25 @@ namespace ranges
             view_interface &operator=(view_interface &&) = default;
             view_interface &operator=(view_interface const &) = default;
             // A few ways of testing whether a range can be empty:
-            CONCEPT_REQUIRES(Cardinality >= 0)
+            CONCEPT_REQUIRES(Cardinality >= 0 || Cardinality == infinite)
```

Comment:
This is a behavioral change as well: assume ranges with `infinite` cardinality are never empty.